tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -9605,6 +9605,41 @@ public function testGeneralizeScopeRecursiveType(
 		);
 	}
 
+	public function dataExtractVariables(): array
+	{
+		return [
+			[
```

Comment:
Hi, this is a nice contribution. Can you please rewrite the tests by looking at NodeScopeResolverTest::testFileAsserts() in this file?

Reply:
If cause I will, thanks for the kind feedback and hints.

Reply:
done, and much cleaner. Thanks for the hint